workers/test_env.go

```diff
@@ -81,16 +86,15 @@ func (env *TestEnvironment) NexusEndpointName() string {
 }
 
 func SetupTestEnvironment(t *testing.T, opts ...TestEnvOption) *TestEnvironment {
-	logger := zaptest.NewLogger(t).Sugar()
```

Comment:
Moved the logger into `RunExecutorTest` so it's per test execution instead of per test setup which makes sure the logs are associated with the right test now.